core/ir/opnd_api.h

```diff
@@ -1703,6 +1703,12 @@ typedef enum _dr_opnd_flags_t {
      * representing their element size.
      */
     DR_OPND_IS_VECTOR = 0x100,
+    /**
+     * Predicate registers can either be merging, zero or neither. If one of these
+     * are set then they are either a merge or zero otherwise aren't either.
```

Comment:
There are plenty of instructions which write to or read from predicate registers rather than apply them- in those cases it doesn't make sense to tag the predicate as merging or zero as we're just treating it as a regular register. Ptest, in this patch's tests,  is an example.

Reply:
Makes sense. So for vector ops they have to be either merging or zero.